packages/web-components/src/utils/date-utils.ts

```diff
@@ -131,11 +132,72 @@ export const days = {
   12: thirtyOneDays,
 };
 
-export function checkLeapYear(year) {
+/**
+ * Return true if the year is a leap year.
+ * (Exported for testing purposes)
+ */
+export function checkLeapYear(year: number) {
   //three conditions to find out the leap year
   return (0 == year % 4 && 0 != year % 100) || 0 == year % 400;
 }
 
+/**
+ * This is used to validate date components and:
+ * 1. Indicate which field fails the built-in validation
+ * 1. Supply an error message to help resolve the issue
+ *
+ * It relies on the component's mutable props.
+ */
+export function validate(
+  component: Components.VaDate | Components.VaMemorableDate,
+  year: number,
+  month: number,
+  day: number,
+  monthYearOnly : boolean = false) : void {
+
+  const leapYear = checkLeapYear(year);
+  const daysForSelectedMonth = leapYear && month == 2 ? 29 : days[month]?.length || 0;
```

Comment:
Checking for leap years is a bit different as well. Previously we would do that validation as its own thing:

https://github.com/department-of-veterans-affairs/component-library/blob/ff6baa66f08b809d2bf3c4276c3da4f034b3968e/packages/web-components/src/components/va-date/va-date.tsx#L129-L133

But now it's built into `daysForSelectedMonth`